### PR TITLE
User controller

### PR DESCRIPTION
**Pull request by wleoguty91**
## Summary

- Added user controller with crud operations and book operations
- Added the common exceptions for user

## Trello Card

- https://trello.com/c/0mcLSvLI/21-crear-user-controller-con-las-operaciones-crud-de-user
Screenshot from 2019-12-30 12-37-26

### CODE REVIEW

File: src/main/java/wolox/training/controllers/UserController.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import wolox.training.exceptions.UserNotFoundException;
+import wolox.training.models.Book;
+import wolox.training.models.User;
+import wolox.training.repositories.UserRepository;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+
+    private UserRepository userRepository;
```
Comment (facucompe) [inline src/main/java/wolox/training/controllers/UserController.java:26]: You can add `@Autowired` to repositories too

File: src/main/java/wolox/training/controllers/UserController.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+    @Autowired
+    public UserController(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
```
Comment (facucompe) [inline src/main/java/wolox/training/controllers/UserController.java:31]: Remove this

File: src/main/java/wolox/training/controllers/UserController.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+    @PutMapping("/{id}")
+    public User updateUser(@RequestBody User user, @PathVariable Long id)
+        throws UserIdDontMatchException, UserNotFoundException {
+        if (user.getId() != id) {
+            throw new UserIdDontMatchException();
+        }
+        userRepository.findById(id).orElseThrow(UserNotFoundException::new);
+        return userRepository.save(user);
```
Comment (facucompe) [inline src/main/java/wolox/training/controllers/UserController.java:52]: Are you updating user info?

**Review by facucompe — CHANGES_REQUESTED** [review]

File: src/main/java/wolox/training/controllers/UserController.java (hunk 1 above)
Comment (leonplondon) [reply src/main/java/wolox/training/controllers/UserController.java:26]: He is using it at constructor level in favor of constructor-based injection

File: src/main/java/wolox/training/controllers/UserController.java (hunk 3 above)
Comment (wleoguty91) [reply src/main/java/wolox/training/controllers/UserController.java:52]: Yes, I do!
Screenshot from 2020-01-03 13-13-58

File: src/main/java/wolox/training/controllers/UserController.java (hunk 2 above)
Comment (wleoguty91) [reply src/main/java/wolox/training/controllers/UserController.java:31]: I applied the suggestion that make @leonplondon , I use dependency injection

File: src/main/java/wolox/training/exceptions/UserNotFoundException.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package wolox.training.exceptions;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(code = HttpStatus.NOT_FOUND, reason = "User not found")
+public class UserNotFoundException extends Exception {
+
+    public UserNotFoundException() {
+        super();
```
Comment (leonplondon) [inline src/main/java/wolox/training/exceptions/UserNotFoundException.java]: Same as previous PRs

Comment (wleoguty91) [reply src/main/java/wolox/training/exceptions/UserNotFoundException.java]: Uppps, a rebase don't work that I expect :sweat_smile:

**Review by facucompe — APPROVED** [review]

**Review by leonplondon — APPROVED** [review]

**Pull request merged**